pep-0613.rst

```diff
@@ -0,0 +1,223 @@
+PEP: 613
+Title: Explicit Type Aliases
+Author: Shannon Zhu <szhu@fb.com>
+Sponsor: Guido van Rossum <guido@python.org>
+Discussions-To: https://mail.python.org/archives/list/typing-sig@python.org/thread/MWRJOBEEEMFVXE7CAKO7B4P46IPM4AN3/
+Status: Draft
+Type: Informational
+Content-Type: text/x-rst
+Created: 21-Jan-2020
+Post-History: 21-Jan-2020
+
+
+Abstract
+========
+
+Type aliases are user-specified types which may be as complex as any type hint,
+and are specified with a simple variable assignment on a module top level.
+
+This PEP formalizes a way to explicitly declare an assignment as a type alias.
+
+Motivation
+==========
+
+Type aliases are declared as top level variable assignments.
+In `PEP 484 <https://www.python.org/dev/peps/pep-0484/#type-aliases/>`_,
+the distinction between a valid type alias and a global variable was implicitly
+determined: if a top level assignment is unannotated, and the assigned value is
+a valid type, then the name being assigned to is a valid type alias. Otherwise,
+that name is simply a global value that cannot be used as a type hint.
+
+These implicit type alias declaration rules create confusion when type aliases
+involve forward references, invalid types, or violate other restrictions
+enforced on type alias declaration. Because the distinction between an
+unannotated value and a type alias is implicit, ambiguous or incorrect type
+alias declarations implicitly default to a valid value assignment. This creates
+expressions that are impossible to express as type aliases and punts error
+diagnosis of malformed type aliases downstream.
+
+The following examples each include an illustration of some of the suboptimal
+or confusing behaviors resulting from existing implicit alias declarations.
+We also introduce explicit aliases of the format :code:`TypeName: TypeAlias = Expression`
+here for the sake of comparison, but the syntax is discussed in further detail
+in later sections.
+
+Forward References:
+*******************
+
+::
+
+  MyType = "ClassName"
+  def foo() -> MyType: ...
+
+This code snippet should not error so long as :code:`ClassName` is defined
+later on. However, a type checker is forced to throw spurious errors that
+(1) :code:`MyType` is an unannotated global string, and (2) :code:`MyType`
```

Comment:
reworded to 'may' rather than specifying any behavior typecheckers must take here - thought about this one for a while and want to keep both statements about potential type checker behavior in for clarity

Reply:
Makes sense!